SpatialGDK/Source/SpatialGDKTests/SpatialGDK/EngineClasses/GridBasedLBStrategy/GridBasedLBStrategyTest.cpp

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "CoreMinimal.h"
+#include "Engine/World.h"
+#include "EngineClasses/GridBasedLBStrategy.h"
+#include "GameFramework/DefaultPawn.h"
+#include "GameFramework/GameStateBase.h"
+#include "SpatialConstants.h"
+#include "TestDefinitions.h"
+#include "TestGridBasedLBStrategy.h"
+#include "Tests/AutomationCommon.h"
+#include "Tests/AutomationEditorCommon.h"
+
+#define GRIDBASEDLBSTRATEGY_TEST(TestName) \
+	TEST(Core, UGridBasedLBStrategy, TestName)
+
+UWorld* TestWorld;
+TMap<FName, AActor*> TestActors;
+UGridBasedLBStrategy* Strat;
+
+// Copied from AutomationCommon::GetAnyGameWorld()
+UWorld* GetAnyGameWorld()
+{
+	UWorld* World = nullptr;
+	const TIndirectArray<FWorldContext>& WorldContexts = GEngine->GetWorldContexts();
+	for (const FWorldContext& Context : WorldContexts)
+	{
+		if ((Context.WorldType == EWorldType::PIE || Context.WorldType == EWorldType::Game)
+			&& (Context.World() != nullptr))
+		{
+			World = Context.World();
+			break;
+		}
+	}
+
+	return World;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND(FCleanup);
+bool FCleanup::Update()
+{
+	TestWorld = nullptr;
+	TestActors.Empty();
+	Strat = nullptr;
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_FIVE_PARAMETER(FCreateStrategy, uint32, Rows, uint32, Cols, float, WorldWidth, float, WorldHeight, uint32, LocalWorkerIdIndex);
+bool FCreateStrategy::Update()
+{
+	Strat = UTestGridBasedLBStrategy::Create(Rows, Cols, WorldWidth, WorldHeight);
+	Strat->Init(nullptr);
+
+	TArray<uint32> VirtualWorkerIds = Strat->GetVirtualWorkerIds();
+	Strat->SetLocalVirtualWorkerId(VirtualWorkerIds[LocalWorkerIdIndex]);
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND(FWaitForWorld);
+bool FWaitForWorld::Update()
+{
+	TestWorld = GetAnyGameWorld();
+
+	if (TestWorld && TestWorld->AreActorsInitialized() )
+	{
+		AGameStateBase* GameState = TestWorld->GetGameState();
+		if (GameState && GameState->HasMatchStarted() )
+		{
+			return true;
+		}
+	}
+
+	return false;
```

Comment:
This should presumably fail the test its in before it tries to do more stuff.

Reply:
Had misunderstood exactly what latent commands do. We could make this fail on a timeout, but it returning false here is not a reason to fail.